### PR TITLE
[1/2] Expose dynamic partition requests via GQL for DryRunInstigationTick

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation

Exposes dynamic partition requests via graphql when testing sensors

## How I Tested These Changes

pytest

### CODE REVIEW

File: python_modules/dagster-graphql/dagster_graphql/schema/instigation.py [hunk 1]
```diff
@@ -267,11 +299,14 @@ def resolve_evaluationResult(self, graphene_info: ResolveInfo):
 
 
 class GrapheneTickEvaluation(graphene.ObjectType):
+    dynamicPartitionRequests = graphene.List(graphene.NonNull(GrapheneDynamicPartitionRequest))
```
Comment (clairelin135) [inline python_modules/dagster-graphql/dagster_graphql/schema/instigation.py:305]: nit: dynamicPartitionRequests -> dynamicPartitionsRequests to mirror `execution_data.dynamic_partitions_requests`

File: python_modules/dagster-graphql/dagster_graphql/schema/instigation.py [hunk 2]
```diff
@@ -127,6 +131,34 @@ class Meta:
     hasMore = graphene.NonNull(graphene.Boolean)
 
 
+class GraphenePartitionRequestType(graphene.Enum):
+    ADD_PARTITIONS = "ADD_PARTITIONS"
+    DELETE_PARTITIONS = "DELETE_PARTITIONS"
+
+
+class GrapheneDynamicPartitionRequest(graphene.ObjectType):
```
Comment (clairelin135) [inline python_modules/dagster-graphql/dagster_graphql/schema/instigation.py:142]: nit: `GrapheneDynamicPartitionRequest` -> `GrapheneDynamicPartitionsRequest`

**Review by clairelin135 — APPROVED** [review]
LGTM! Added a few small naming nits

**Comment by salazarm** [comment]
build failures seem unrelated

**Pull request merged**